ci: no job sets timeout-minutes, so a hung build burns 6 hours before failing

No workflow job sets `timeout-minutes`, so a hung job runs to GitHub's 6-hour default instead of failing fast. This is not theoretical — it cost roughly 18 hours of runner time in a single afternoon on 2026-08-04.

## What happens

The `linux/arm64` leg of the image build runs `npm ci --ignore-scripts` under QEMU user-mode emulation. QEMU intermittently crashes there:

```
#34 [linux/arm64 ui 4/6] RUN npm ci --ignore-scripts
#34 20.86 qemu: uncaught target signal 4 (Illegal instruction) - core dumped
```

BuildKit does **not** fail fast on this. The crashed emulated child leaves the parent waiting, the step emits no further output, and the job simply sits there until GitHub force-cancels it:

```
##[error]The operation was canceled.
```

The `ci` gate then correctly reports failure on `build: cancelled` — so the required check behaves properly. The problem is purely how long it takes to get there.

## Evidence

| Run | Duration | Outcome |
|---|---|---|
| [30933449722](https://github.com/jacaudi/tempestwx-utilities/actions/runs/30933449722) (#113) | **6h02m** | cancelled |
| [30933562939](https://github.com/jacaudi/tempestwx-utilities/actions/runs/30933562939) (#93) | **6h03m** | cancelled |
| [30933778452](https://github.com/jacaudi/tempestwx-utilities/actions/runs/30933778452) (#74) | ~5h58m on the build job | cancelled |

For comparison, healthy `main` runs of the same workflow complete in **16–28 minutes**.

## It is a flake, not a real incompatibility

Confirmed rather than assumed:

- `linux/amd64` (native, no emulation) completed normally in the same runs.
- PR #100, which builds the **same** arm64 QEMU path with a *newer* Node major, succeeded in ~11.5 minutes minutes later on the same infrastructure.
- Re-running #74 unchanged passed cleanly — same commit, same code.

So the failing builds were not caused by the dependency under test in any of the three cases. Each one was a Renovate PR that got misdiagnosed as "this bump breaks the build" until the log was read.

## Cost

Three jobs × ~6 hours = ~18 hours of runner time producing no signal, plus the review time spent diagnosing dependency bumps that were never at fault.

## Proposed fix

Add `timeout-minutes` to the `build` job in `.github/workflows/ci-build.yml` (job defined at line 54). Something around **45** leaves comfortable headroom over the observed 16–28 minute legitimate multi-arch build while failing a hang in under an hour instead of six.

Worth considering for the other jobs too — `ci-smoke.yml`'s `smoke` job (line 24) polls a container for up to 60s and could likewise hang on a bad image, and `ci-test.yml`'s `test` job (line 17) now installs the full scientific Python stack.

Optional follow-up if the flake recurs: pin or bump `docker/setup-qemu-action`'s QEMU version, which is where the SIGILL originates.

## Note on scope

The same gap exists in the upstream `github-actions-align` templates these workflows were copied from, so fixing it here is worth mirroring there — otherwise the next repo onboarded inherits it.

--- a/.github/workflows/ci-build.yml
+++ b/.github/workflows/ci-build.yml
@@ -54,6 +54,11 @@ jobs:
   build:
     name: build + push image
     runs-on: ubuntu-latest
+    # The arm64 leg runs under QEMU, which can crash (SIGILL) without failing
+    # the step — BuildKit then waits forever. Healthy builds finish in ~11-14
+    # min, so 45 is ample headroom while capping a hang well under GitHub's
+    # 6-hour default.
+    timeout-minutes: 45
     permissions:
       contents: read
       packages: write

--- a/.github/workflows/ci-release.yml
+++ b/.github/workflows/ci-release.yml
@@ -34,6 +34,7 @@ jobs:
   release:
     name: release-please
     runs-on: ubuntu-latest
+    timeout-minutes: 10
     permissions:
       contents: write
       pull-requests: write

--- a/.github/workflows/ci-smoke.yml
+++ b/.github/workflows/ci-smoke.yml
@@ -24,6 +24,8 @@ jobs:
   smoke:
     name: smoke test image
     runs-on: ubuntu-latest
+    # Polls the container for up to 60s; a bad image could otherwise hang here.
+    timeout-minutes: 10
     permissions:
       contents: read
       packages: read

--- a/.github/workflows/ci-test.yml
+++ b/.github/workflows/ci-test.yml
@@ -17,6 +17,8 @@ jobs:
   test:
     name: ${{ inputs.task }}
     runs-on: ubuntu-latest
+    # Observed 2-4 min; the headroom covers the scientific Python install.
+    timeout-minutes: 20
     services:
       # The Postgres-backed tests in internal/postgres are guarded on
       # POSTGRES_URL and skip without it, so without this the Postgres half of

--- a/.github/workflows/ci.yaml
+++ b/.github/workflows/ci.yaml
@@ -29,6 +29,7 @@ jobs:
   context:
     name: context
     runs-on: ubuntu-latest
+    timeout-minutes: 5
     permissions:
       contents: read
       pull-requests: read
@@ -147,6 +148,7 @@ jobs:
     needs: [context, test, build, smoke]
     if: ${{ !cancelled() }}   # required: else this is SKIPPED on failure, which can satisfy protection
     runs-on: ubuntu-latest
+    timeout-minutes: 5
     steps:
       - name: Assert required stages
         env:

--- a/.github/workflows/release-republish.yml
+++ b/.github/workflows/release-republish.yml
@@ -29,6 +29,7 @@ jobs:
     name: retag ${{ inputs.tag }}
     if: ${{ inputs.mode == 'retag' }}
     runs-on: ubuntu-latest
+    timeout-minutes: 15
     permissions:
       contents: read
       packages: write

